### PR TITLE
Http server

### PR DESCRIPTION
**Pull request by ceolin**
Add support to create http server. This series introduce code to make possible to create a basic http server and node types for some of our primitive types, such as irange, boolean and string.

It's necessary libmicrohttpd (https://www.gnu.org/software/libmicrohttpd/) to build it.

### CODE REVIEW

**Comment by cmarcelo** [comment]
Is there something we could do to automatically test the `sol_http_server`? Maybe a test case that spawn a few clients that would use `sol_http_client` and await for requests?

File: src/lib/comms/Makefile [hunk 1]
```diff
@@ -18,12 +18,21 @@ obj-networking-$(OIC) += \
...
     sol-http-client-impl-curl.o
 
 obj-networking-$(HTTP_CLIENT)-extra-ldflags += \
     $(LIBCURL_LDFLAGS)
 
+obj-networking-$(HTTP_SERVER) += \
+    sol-http-server-impl-microhttpd.o
+
+obj-networking-$(HTTP_SERVER)-extra-ldflags += \
+    $(LIBMICROHTTPD_LDFLAGS)
```
Comment (dorileo) [inline src/lib/comms/Makefile]: no cflags required?

File: src/samples/flow/http-server/boolean.fbp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# Reads key 'a' (code 97) from keyboard (stdin) and toggles state,
+
```
Comment (dorileo) [inline src/samples/flow/http-server/boolean.fbp]: maybe add these samples to be built as well? I mean, if you follow the other flow samples pattern they will be get the C code generated and built by the ```make check``` target.

**Comment by ceolin** [comment]
fixes on https://github.com/solettaproject/soletta/pull/686

**Pull request closed without merging**